File: 3_bruteforce_backtracking/이준우/소수 찾기.py
import math
from itertools import permutations


def is_prime(n):
    if n < 2:
        return False

    for i in range(2, int(math.sqrt(n)) + 1):
        if n % i == 0:
            return False
    return True


def solution(numbers):
    answer = 0
    n_set = set()

    for i in range(1, len(numbers) + 1):
        for j in permutations(list(numbers), i):
            n_set.add(int("".join(j)))

    for num in list(n_set):
        answer += is_prime(num)

    return answer

File: 3_bruteforce_backtracking/김민주/소수찾기.java
import java.util.*;

class Solution {
    Set<Integer> numberSet= new HashSet<>();
    
    public int solution(String numbers) {
        boolean[] visited = new boolean[numbers.length()];
        
        dfs(numbers, "", visited);
        
        int answer = 0;
        for (int num :numberSet) {
            if (isPrime(num)) {
                answer++;
            }
        }
        
        return answer; 
    }
    
    // 백트래킹
    void dfs(String numbers, String current, boolean[] visited) {
        if (!current.equals("")) {
            numberSet.add(Integer.parseInt(current));
        }
        
        for (int i = 0; i < numbers.length(); i++) {
            if (!visited[i]) {
                visited[i] = true;
                dfs(numbers, current + numbers.charAt(i), visited);
                visited[i] = false; 
            }
        }
    }
    
    // 소수 판별 함수
    boolean isPrime(int num) {
        if (num < 2) return false;
        
        for (int i = 2; i <= Math.sqrt(num); i++) {
            if (num % i == 0) return false;
        }
        
        return true;
    }
}